src/uu/dircolors/src/dircolors.rs

```diff
@@ -138,15 +138,26 @@ pub fn uumain(args: impl uucore::Args) -> UResult<()> {
         let fin = BufReader::new(std::io::stdin());
         result = parse(fin.lines().filter_map(Result::ok), &out_format, files[0]);
     } else {
-        match File::open(files[0]) {
+        let path = Path::new(files[0]);
+        if path.is_dir() {
+            return Err(USimpleError::new(
+                2,
+                format!("expected file, got directory {}", path.quote()),
+            ));
+        }
+        match File::open(path) {
             Ok(f) => {
                 let fin = BufReader::new(f);
-                result = parse(fin.lines().filter_map(Result::ok), &out_format, files[0]);
+                result = parse(
+                    fin.lines().filter_map(Result::ok),
```

Comment:
This is an instance of the clippy lint we proposed :)

Reply:
To clarify a bit, this should be `map_while`, even though we don't get the infinite loop anymore on a directory, it would still be safer to stop on an error.